### PR TITLE
Fix accuracy function on chapter 3.6

### PR DESCRIPTION
**Pull request by hsneto**
Hello there,

I got an error while running the code `d2l.train_ch3(net, train_iter, test_iter, loss, num_epochs, trainer).` Further examination showed to me that the error was caused by the `accuracy` function trying to compare a int64 with a float32.

Below is my change to fix this issue:
(added a `.astype('float32')`)

```python
def accuracy(y_hat, y):
    if y_hat.shape[1] > 1:
        return float((y_hat.argmax(axis=1).astype('float32') == y.astype('float32')).sum())
    else:
        return float((y_hat.astype('int32') == y.astype('int32')).sum())
```

### CODE REVIEW

**Comment by astonzhang** [comment]
Can you let me know which notebook has the code 
`d2l.train_ch3(net, train_iter, test_iter, loss, num_epochs, trainer)` that got the error?

**Comment by hsneto** [comment]
@astonzhang , this problem occurs on the notebooks below:

- chapter_linear-networks/softmax-regression-gluon.ipynb
- chapter_multilayer-perceptrons/dropout.ipynb
- chapter_multilayer-perceptrons/mlp-gluon.ipynb

**Comment by astonzhang** [comment]
Running ok on my end, can you make sure the correct version of mxnet is installed:
https://d2l.ai/chapter_installation/index.html

Do you use windows or macos/linux?

**Comment by hsneto** [comment]
```
$ uname
$ python3 -V
$ pip list | grep mxnet
```
Linux
Python 3.5.2
mxnet-cu102mkl       1.6.0b20200215

**Comment by astonzhang** [comment]
Thanks. Will use astype('float32')
